doc/dev/development.md

```diff
@@ -1,327 +1,7 @@
 # Development Guide
```

Comment:
I'd prefer that we delete this document and update the root-level README.md to point developers to istio/istio directly.